BH/Modules/Item/ItemDisplay.cpp

```diff
@@ -1778,7 +1783,7 @@ bool MagicPrefixCondition::EvaluateInternalFromPacket(ItemInfo* info,
 		return false;
 	}
 
-	return IntegerCompare(-1, operation, prefixID);
+	return IntegerCompare(-1, operation, prefixID1);

```

Comment:
Should this be `return IntegerCompare(-1, operation, prefixID1, prefixID2);`, at least to match the MagicSuffixCondition::EvaluateInternalFromPacket? I also don't quite understand the point of this last check, comparing against -1 should always fail, right?

Reply:
Yeah good catch, looks like I just forgot to add it. And TBH I'm not sure I understand the point of _any_ of the `EvaluateInternalFromPacket` calls. All of the things that I've played around with so far seemed to have only used `EvaluateInternal`.

As for the -1, I think that was the intention yeah (or maybe just a placeholder? /shrug) It probably makes sense to have these just `return false` instead

Reply:
So I was wrong. It _is_ definitely going through the `EvaluateInternalFromPacket` code path too. But even though it always returns false, the filter still works correctly. I could add the same logic that's in `EvaluateInternal` but if it functions the same without it then I'm not entirely sure what the point is, or if it's even worth doing that

Reply:
Actually it doesn't even matter because `uInfo->item->pItemData->wPrefix[0]` != `info->prefixes[0]` (`uInfo` has one continuous list of suffix+prefix+automod but `info` appears to keep separate lists).

Merged updates so this should be good to go now.